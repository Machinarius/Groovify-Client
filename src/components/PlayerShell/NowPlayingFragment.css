.nowplaying-container {
    background: #282828;

    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;

    border-color: #1ed760;
    border-width: 1px;
    border-style: solid none none none;
}

.info-container {
    align-self: start;
    width: 25vw;
}

.padding-element {
    height: 90px;
    align-self: end;
    width: 25vw;
}

.controls-container {
    flex-grow: 1;
    align-self: center;
}

.buttons-container {
    display: flex;
    justify-content: center;
}

.song-info {
    display: flex;
    flex-direction: row;
    justify-content: start;
    align-items: center;
}

.current-cover {
    width: 60px;
    height: 60px;
    margin-left: 1em;
    margin-bottom: 15px;
    margin-top: 15px;
    margin-right: 0.5em;
}

.song-text-container > * { 
    margin: 0;
}

.progress-container {
    display: flex;
    flex-direction: row;
    align-items: center;
}

.playback-progress {
    flex-grow: 1;
    height: 1em;
}

.playback-position {
    align-self: flex-start;
    margin: 0;
    margin-right: 0.25em;
}

.playback-length {
    margin: 0;
    margin-left: 0.25em;
}