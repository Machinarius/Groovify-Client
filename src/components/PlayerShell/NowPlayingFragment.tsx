import * as React from "react";

import IAuthenticationService from "../../services/IAuthenticationService";
import IMusicPlayerController from "../../services/IMusicPlayerController";

export interface IProps {
    authService: IAuthenticationService,
    playerController: IMusicPlayerController
}

export default class NowPlayingFragment extends React.Component<IProps, {}> {
    private authService: IAuthenticationService;

    constructor(props: IProps) {
        super(props);

        this.authService = props.authService;
    }

    render() {
        return (<div></div>);
    }
}