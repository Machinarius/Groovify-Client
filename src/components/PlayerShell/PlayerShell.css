.main-stack {
    display: flex;
    flex-direction: column;

    width: 100%;
    height: 100%;
}

.profile-container {
    width: 100%;
}

.library-container {
    flex-grow: 1;
    overflow-x: hidden;
    overflow-y: scroll;
}
