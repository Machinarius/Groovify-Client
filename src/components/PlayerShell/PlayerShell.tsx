import * as React from "react";

export default class PlayerShell extends React.Component {
    render() {
        return (<div></div>);
    }
}