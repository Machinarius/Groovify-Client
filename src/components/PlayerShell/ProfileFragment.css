.profile-container {
    background: #282828;

    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;
}

.app-logo {
    align-self: flex-start;
    flex-grow: 1;
    margin-left: 1em;
}

.profile-text {
    margin: 0 1em 0 0;
}

.profile-picture {
    width: 60px;
    height: 60px;
    border-radius: 50%;
    margin-right: 1em;
    margin-bottom: 0.25em;
    margin-top: 0.25em;
}

.profile-name {
    margin-bottom: 0;
    margin-top: 0;
}

.logout-button {
    text-align: right;
    display: block;
}