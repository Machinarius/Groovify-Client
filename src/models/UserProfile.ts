export default interface UserProfile {
    name: string;
    pictureUrl: string;
}