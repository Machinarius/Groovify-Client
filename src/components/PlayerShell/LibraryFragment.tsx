import * as React from "react";

import IAuthenticationService from "../../services/IAuthenticationService";
import IMusicPlaybackController from "../../services/IMusicPlaybackController";
import IMusicLibraryRepository from "../../services/IMusicLibraryRepository";

import GroovifyAPIMusicLibraryRepository from "../../services/GroovifyAPIMusicLibraryRepository";

import { Album } from "../../models/Album";
import AlbumComponent from "./Library/AlbumComponent";

export interface IProps {
    authService: IAuthenticationService,
    playbackController: IMusicPlaybackController,
    overrideLibraryRepo?: IMusicLibraryRepository
}

interface IState {
    didLoadAllAlbums: boolean,
    albums: Album[]
}

export default class LibraryFragment extends React.Component<IProps, IState> {
    private authService: IAuthenticationService;
    private playbackController: IMusicPlaybackController;
    private libraryRepo: IMusicLibraryRepository;

    constructor(props: IProps) {
        super(props);

        this.authService = props.authService;
        this.playbackController = props.playbackController;
        this.libraryRepo = props.overrideLibraryRepo ||
            new GroovifyAPIMusicLibraryRepository();

        this.initializeAsync = this.initializeAsync.bind(this);
        this.handlePlaybackRequested = this.handlePlaybackRequested.bind(this);

        this.state = {
            didLoadAllAlbums: false,
            albums: null
        };
    }

    componentDidMount() {
        this.initializeAsync();
    }

    private async initializeAsync(): Promise<any> {
        let allAlbums = await this.libraryRepo.getAllAlbumsAsync();
        this.setState({
            albums: allAlbums,
            didLoadAllAlbums: true
        });
    }

    private handlePlaybackRequested(songId: string, albumId: string): void {
        this.playbackController.startPlayback(songId, albumId);
    }

    render() {
        if (this.state.didLoadAllAlbums) {
            if (this.state.albums.length) {
                var albumComponents = this.state.albums.map(album => 
                    <li key={album.id}>
                        <AlbumComponent albumObject={album} onPlaybackRequested={this.handlePlaybackRequested}></AlbumComponent>
                    </li>);

                return (
                    <ul style={{listStyleType: "none"}}>{albumComponents}</ul>
                );
            }

            return (
                <div>
                    <h1 className="loading-text">No albums have been added to our library yet, please check back later!</h1>
                </div>
            );
        }

        return (
            <div>
                <p className="loading-text">Loading albums...</p>
            </div>
        );
    }
}