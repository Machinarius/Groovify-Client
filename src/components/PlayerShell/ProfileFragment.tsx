import * as React from "react";

import IAuthenticationService from "../../services/IAuthenticationService";
import UserProfile from "../../models/UserProfile";

export interface IProps {
    authService: IAuthenticationService
}

interface IState {
    loadingProfileData: boolean;
    userProfile: UserProfile
}

export default class ProfileFragment extends React.Component<IProps, IState> {
    private authService: IAuthenticationService;

    constructor(props: IProps) {
        super(props);

        this.loadProfileData = this.loadProfileData.bind(this);
        this.logOut = this.logOut.bind(this);

        this.authService = props.authService;
        this.state = {
            loadingProfileData: true,
            userProfile: undefined
        };
    }

    componentDidMount() {
        this.loadProfileData();
    }

    private async loadProfileData(): Promise<any> {
        let profile = await this.authService.getProfileAsync();
        this.setState({
            loadingProfileData: false,
            userProfile: profile
        });
    }

    private logOut(): void {
        this.authService.logOut();
    }

    render() {
        if (this.state.loadingProfileData) {
            return (<p className={"loading-message"}>Loading data...</p>);
        } 
        
        return (
            <div>
                <p className={"profile-name"}>Welcome, {this.state.userProfile.name}</p>
                <img className={"profile-picture"} src={this.state.userProfile.pictureUrl} />
                <button className={"logout-button"} onClick={() => this.logOut()}>Log Out</button>
            </div>
        );
    }
}