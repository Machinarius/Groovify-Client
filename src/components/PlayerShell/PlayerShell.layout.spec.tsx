jest.mock("./LibraryFragment");
jest.mock("./ProfileFragment");
jest.mock("./PlayerShell");

import { mock, instance, when } from "ts-mockito"

import * as React from "react";
import * as testRenderer from "react-test-renderer";

import IAuthenticationService from "../../services/IAuthenticationService";

import UserProfile from "../../models/UserProfile";

import PlayerShell from "./PlayerShell";

test("Must display all child components with correct CSS classes", () => {
    let fakeProfile: UserProfile = {
        name: "Germ√°n Valencia",
        pictureUrl: "https://giphy.com/gifs/Px8HAmJdeiiIw/html5"
    };

    // TODO: Use enzyme shallow rendering with snapshots to avoid having to setup this promise
    // https://stackoverflow.com/questions/55341289/configure-enzyme-to-json-with-jest
    let fakeProfilePromise = new Promise<UserProfile>(resolve => resolve(fakeProfile));
    let mockAuthService = mock<IAuthenticationService>();
    when(mockAuthService.getProfileAsync()).thenReturn(fakeProfilePromise);

    let authService = instance(mockAuthService);
    let component = testRenderer.create(
        <PlayerShell authService={authService} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});