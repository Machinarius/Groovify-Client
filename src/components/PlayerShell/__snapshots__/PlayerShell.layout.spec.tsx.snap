// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Must display all child components with correct CSS classes 1`] = `null`;
